### DISCUSSION TITLE
Find all nodes on specific level

### DISCUSSION THREAD

**Discussion by forthrin** [sparklemotion/nokogiri#3138] [General Q&A] [answered]
Given an unordered list with multiple levels (lists within lists), how does one find only the list elements only on the first level? (Or at any given level.) `html.css('li')` will find all list elements on *all* levels, and it seems there is no CSS rule to stop cascading.

**Comment by flavorjones** [comment, marked as answer]
@forthrin I generally reach for [XPath queries](https://www.w3.org/TR/1999/REC-xpath-19991116/) if I'm doing something more complicated than a tag name or an attribute.

Try `doc.xpath("//li[not(ancestor::li)]")` and if that doesn't do what you want, please provide a sample input and your expected output?

**Comment by forthrin** [comment]
Yep, that works. Just wondered if there was a CSS way to do it. Thanks!
